File: c/SortArrayByParity.c
// https://leetcode.com/problems/sort-array-by-parity/

int* sortArrayByParity(int* nums, int numsSize, int* returnSize){
    for (int i = 0, j = 0; i < numsSize; i++) {
        int temp;
        if (nums[i] % 2 == 0 && nums[j] % 2 != 0) {
            temp = nums[i];
            nums[i] = nums[j];
            nums[j++] = temp;
        } else if (nums[i] % 2 == 0 && nums[j] % 2 == 0) {
            j++;
        }
    }
    *returnSize = numsSize;
    return nums;
}

File: swift/SortArrayByParity.swift
// https://leetcode.com/problems/sort-array-by-parity/

class Solution {
    func sortArrayByParity(_ nums: [Int]) -> [Int] {
        var nums = nums
        var lastEven = 0 
        for i in 0 ..< nums.count {
            if nums[i] % 2 == 0 {
                nums.swapAt(i, lastEven)
                lastEven += 1
            }
        }
        return nums
    }
}